com.microsoft.mrtk.accessibility/Subsystems/MRTK/MRTKAccessibilitySubsystem.cs

```diff
@@ -40,10 +41,157 @@ class AccessibilityProvider : Provider
+                List<GameObject> objectList)
+            {
+                if (maxDistance <= 0)
+                {
+                    Debug.LogError("The distance from the camera to the objects cannot be less than or equal to zero.");
+                    return false;
+                }
+
+                AssembleDescribableObjects(classifications, objectList);
+                FilterDescribableObjects(readerView, maxDistance, objectList);
+
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryRegisterDescribableObject(
+                GameObject gameObj,
+                ObjectClassification classification)
+            {
+                // Make sure the specified game object has not been previously registered in any classification.
+                foreach (IList<GameObject> list in describableObjects.Values)
+                {
+                    if (list.Contains(gameObj))
+                    {
+                        Debug.LogError($"{gameObj.name} has already been registerd as a describable object");
+                        return false;
+                    }
+                }
+
+                describableObjects[classification].Add(gameObj);
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryUnregisterDescribableObject(
+                GameObject gameObj,
+                ObjectClassification classification)
+            {
+                List<GameObject> objCollection = describableObjects[classification];
+                if (!objCollection.Contains(gameObj))
+                {
+                    Debug.LogError($"{gameObj.name} has not been registerd as a describable object of classification {classification}");
+                    return false;
+                }
+
+                if (!describableObjects[classification].Remove(gameObj))
+                {
+                    Debug.LogError($"Failed to unregister {gameObj.name} as a describable object of classification {classification}");
+                    return false;
+                }
+
+                return true;
+            }
+
+            /// <summary>
+            /// Collects the registered <see cref="GameObject"/>s from the requested <see cref="ObjectClassification"/>s.
+            /// </summary>
+            /// <param name="classifications">The combined flags specifiying the desired classification(s) (people, places, things, etc.).</param>
+            /// <param name="objectList">The collection which will receive the requested <see cref="GameObject"/>s.</param>
+            /// <remarks>When this method is called, the objectList will be cleared prior to adding the requested <see cref="GameObject"/>s.</remarks>
+            private void AssembleDescribableObjects(
+                ObjectClassification classifications,
+                List<GameObject> objectList)
+            {
+                objectList.Clear();
```

Comment:
Secondhand effects like objectList.Clear make me slightly wary of not using the `out GameObject[] objectList` pattern. Is it possible to refactor the code to use this pattern instead of modifying data structures in place?

Reply:
anything is possible 😄

my concern is the amount of allocations that are likely to occur if we return arrays instead of filling/editing user (read: screen reader) provided lists.

Reply:
you bring up a good point about modifying _user_ data.... this API is one I am strongly considering making internal to the assembly

Reply:
As long as it's documented that the method clears the list, this is a somewhat common pattern in Unity (taking in a list, clearing it, and filling it):

https://docs.unity3d.com/ScriptReference/SubsystemManager.GetSubsystems.html

https://docs.unity3d.com/ScriptReference/Mesh.GetVertices.html

https://docs.unity3d.com/ScriptReference/GameObject.GetComponents.html (scroll past the array returning method 😜)

Reply:
+1 here for modify-list-pattern

Reply:
i will ensure that the api docs are clear